npm/util/ioutil/restore.go

```diff
@@ -264,15 +268,16 @@ func (definition *ErrorDefinition) getErrorLineNumber(stdErr, commandString stri
 // return whether the file was altered
 func (creator *FileCreator) handleLineError(stdErr, commandString string, lineNum int) (bool, *Line) {
 	lineIndex := 0
-	currentLineIndex := 0
+	currentLineNum := 1
 	for i := range creator.lines {
 		if _, isOmitted := creator.lineNumbersToOmit[i]; isOmitted {
 			continue
 		}
-		if currentLineIndex == lineNum-1 {
-			lineIndex = currentLineIndex
+		if currentLineNum == lineNum {
+			lineIndex = i
```

Comment:
bug fix here

Reply:
before, we may map a line error on line 5 to the original line 5 (not the new line 5 after skipping lines on retry). This could lead to using the wrong error handling